webknossos/webknossos/dataset/_utils/buffered_slice_reader.py

```diff
@@ -61,48 +60,18 @@ def __init__(
         self.bbox_current_mag = absolute_bounding_box.in_mag(view.mag)
 
     def _get_slice_generator(self) -> Generator[np.ndarray, None, None]:
-        for batch in get_chunks(
-            list(
-                range(
-                    self.bbox_current_mag.topleft[self.dimension],
-                    self.bbox_current_mag.bottomright[self.dimension],
-                )
-            ),
-            self.buffer_size,
-        ):
-            n_slices = len(batch)
-            batch_start_idx = batch[0]
-
-            assert (
-                n_slices <= self.buffer_size
-            ), f"n_slices should at most be batch_size, but {n_slices} > {self.buffer_size}"
-
-            bbox_offset = self.bbox_current_mag.topleft
-            bbox_size = self.bbox_current_mag.size
-
-            buffer_bounding_box = BoundingBox.from_tuple2(
-                (
-                    bbox_offset[: self.dimension]
-                    + (batch_start_idx,)
-                    + bbox_offset[self.dimension + 1 :],
-                    bbox_size[: self.dimension]
-                    + (n_slices,)
-                    + bbox_size[self.dimension + 1 :],
-                )
-            )
```

Comment:
ND bbox can be two-dimensional, and VecInts can be two-dimensional as well now. It might lead to some errors. Some parts like chunking require three-dimensional data and will fail when there is no z-axis.